akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/javadsl/PersistentEntity.scala

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed.javadsl
+
+import java.util.Optional
+import java.util.function.BiFunction
+import java.util.function.{ Function ⇒ JFunction }
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.BackoffSupervisorStrategy
+import akka.actor.typed.Behavior
+import akka.actor.typed.Props
+import akka.annotation.InternalApi
+import akka.cluster.sharding.ShardCoordinator.ShardAllocationStrategy
+import akka.cluster.sharding.typed.ClusterShardingSettings
+import akka.cluster.sharding.typed.ShardingEnvelope
+import akka.cluster.sharding.typed.ShardingMessageExtractor
+import akka.persistence.typed.javadsl.PersistentBehavior
+
+// FIXME docs
+abstract class PersistentEntity[Command, Event, State >: Null] private (
+  val typeKey:   EntityTypeKey[Command],
+  persistenceId: String, supervisorStrategy: Option[BackoffSupervisorStrategy])
+  extends PersistentBehavior[Command, Event, State](persistenceId: String, supervisorStrategy: Option[BackoffSupervisorStrategy]) {
+
+  def this(typeKey: EntityTypeKey[Command], entityId: String) = {
+    this(typeKey, persistenceId = typeKey.persistenceIdFrom(entityId), None)
+  }
+
+  def this(typeKey: EntityTypeKey[Command], entityId: String, backoffSupervisorStrategy: BackoffSupervisorStrategy) = {
+    this(typeKey, persistenceId = typeKey.persistenceIdFrom(entityId), Some(backoffSupervisorStrategy))
+  }
+
+}
+
+object PersistentEntityRegistry {
+  def get(system: ActorSystem[_]): PersistentEntityRegistry =
+    new PersistentEntityRegistry(ClusterSharding.get(system))
+
+}
+
+class PersistentEntityRegistry(sharding: ClusterSharding) {
```

Comment:
I think bringing the whole concept of registry in is a mistake. Can we instead identify the pain points and make sharding easier to use for them?

Reply:
one concrete pain point is `persistenceId` vs `entityId`, there is no need to care about the `persistenceId` when using sharding, since that should be derived from `entityId`